caikit_computer_vision/modules/segmentation/image_processing.py

```diff
@@ -0,0 +1,281 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Detectron2 postprocess tasks and functions related.
+"""
+# Standard
+from tempfile import NamedTemporaryFile
+from typing import Type, Union
+import json
+
+# Third Party
+from detectron2.evaluation.coco_evaluation import instances_to_coco_json
+from detectron2.modeling.postprocessing import detector_postprocess
+from detectron2.structures import Boxes, Instances
+from pycocotools.coco import COCO
+from tqdm import tqdm
+import cv2
+import numpy as np
+import pycocotools.mask as maskUtils
+import shapely.geometry as spg
+import torch
+
+# First Party
+from caikit.core.exceptions import error_handler
+import alog
+
+log = alog.use_channel("IMAGE_SEGMENTATION")
+error = error_handler.get(log)
+
+
+def maskrcnn_postprocess(
+    results: tuple,
+    uuid: str,
+    image_id: int = 1,
+    threshold=0.2,
+) -> dict:
+    """Postprocess task for MaskRCNN type models.
+
+    Args:
+        results (dict): COCO json from an inference task
+        uuid (str): Unique identifier of image
+        image_id (int, optional): Image ID to store in the task. Defaults to 1.
+
+    Returns:
+        dict: Formatted COCO JSON in OCL format
+    """
+    if len(results) == 0:
+        return {}
+    image_size = results["IMAGES_SIZE"]
+    instances = _convert_results_dict_to_instances(results)
+    log.info(f"Processing Instances of len {len(instances)}")
+    try:
+        result_instances = detector_postprocess(instances, image_size[0], image_size[1])
+    except IndexError as e:
+        log.error(f"Length of MASKS {len(results['MASKS'])}")
+        log.error("No instances found, setting empty annotations", exc_info=e)
+        result_instances = []
+    coco_instances = instances_to_coco_json(result_instances, image_id)
+    coco_json = _assemble_images_coco(
+        file_name=f"{uuid}.jpg",
+        height=image_size[0],
+        width=image_size[1],
+        image_id=image_id,
+    )
+    inf_coco = detectron2_to_coco(
+        detectron2_out=coco_instances,
+        inf_coco=coco_json,
+        score_threshold=threshold,
+        save_coco="",
+    )
+    coco_json = inf_coco.dataset
+    return coco_json
+
+
+def _convert_results_dict_to_instances(results: dict) -> "Instances":
+    """Helper function to convert output of model to Instances.
+
+    Args:
+        results (dict): Raw COCO Dict from Detectron2.
+
+    Returns:
+        Instances: Instances object.
+    """
+    boxes: Boxes = Boxes(results["BOXES"])
```

Comment:
Hopefully  I haven't missed something somewhere - currently the object detection bounding box being used defines them as integers, although I believe COCO uses float boxes, right? I assume this is probably the case for detectron2 also, does this need to be changed?